src/linear_algebra/orientation.lean

```diff
@@ -351,8 +376,25 @@ variables {R} {ι : Type*} [fintype ι] [decidable_eq ι]
 protected def orientation [nontrivial R] (e : basis ι R M) : orientation R M ι :=
 ray_of_ne_zero R _ e.det_ne_zero
 
+lemma orientation_map [nontrivial R] [is_domain R] (e : basis ι R M)
+  (f : M ≃ₗ[R] N) : (e.map f).orientation = orientation.map ι f e.orientation :=
+by simp_rw [basis.orientation, orientation.map_apply, basis.det_map']
+
+-- TODO: does this generalize to an arbitrary orientation, not just `e.orientation`?
+lemma map_orientation_eq_det_inv_smul [nontrivial R] [is_domain R] (e : basis ι R M)
+  (f : M ≃ₗ[R] M) : orientation.map ι f e.orientation = (f.det)⁻¹ • e.orientation :=
```

Comment:
This seems extremely closely related to what I have as `orientation_comp_linear_equiv_eq_iff_det_pos`, so I expect you'll want to refactor that in some way.

Reply:
I'm out of time for today, but I agree; those lemmas at the bottom of the file should probably be expressed in terms of the `orientation.map` I added.

Reply:
Well, they should be *proved* in terms of lemmas for `orientation.map`, I suppose, and start with one stated in terms of the `units` action before deducing the ones about positive and negative determinants, but *statements* for alternating maps still make sense in their own right. All these groups of lemmas are proving related results about different APIs: what (`basis.map`, `orientation.map`, `alternating_map.comp_linear_map`) do to an orientation, in terms of the determinant of the `linear_equiv` involved.

Reply:
I've now refactored so this is proved for an arbitrary orientation and the lemmas at the bottom of the file (special cases when the dimension is used to deduce the existence of a basis, essentially) are restated in terms of `orientation.map` and deduced from this one.